src/services/conversation.py

```diff
@@ -0,0 +1,74 @@
+import os
+from collections.abc import AsyncGenerator
+from typing import Protocol
+
+from agents.common.data import Message
+from agents.graph import IGraph, KymaGraph
+from agents.memory.redis_checkpointer import RedisSaver, initialize_async_pool
+from utils.logging import get_logger
+from utils.models import LLM, ModelFactory
+
+logger = get_logger(__name__)
+
+REDIS_URL = f"{os.getenv('REDIS_URL')}/0"
+
+
+class IService(Protocol):
+    """Service interface"""
+
+    def init_conversation(self) -> dict:
+        """Initialize the conversation"""
+        ...
+
+    """Initialize the conversation"""
+
+    def handle_request(
+        self, conversation_id: int, message: Message
+    ) -> AsyncGenerator[bytes, None]:
+        """Handle a request for a conversation"""
+        ...
+
+
+class SingletonMeta(type):
+    """Singleton metaclass."""
+
+    _instances: dict[type, object] = {}
```

Comment:
Why it needs to be a dict? Can it be a single `_instance: object`?

Reply:
this meta class have class field _instances which can be used by the other classes too. It means it stores a single instances for others.